File: backend/app/schemas/risk.py
from typing import List, Dict, Any, Optional
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from app.models.enums import RiskLevel
from app.schemas.security import FindingResponse


class RiskAssessmentResponse(BaseModel):
    id: Optional[str] = None
    change_id: Optional[str] = None
    risk_level: RiskLevel
    risk_score: float
    reasons: List[str]
    recommended_action: str
    features: Dict[str, float]
    detected_security_findings: List[FindingResponse] = []
    model_info: Dict[str, Any]
    created_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class RiskEvaluationRequest(BaseModel):
    change_id: Optional[str] = None
    raw_content: Optional[str] = None
    plan_json: Optional[Dict[str, Any]] = None


class ModelInfoResponse(BaseModel):
    model_type: str
    features: List[str]
    classes: List[str]
    validation_accuracy: float
    feature_importances: Dict[str, float]
    is_demo: bool
    notice: str

File: frontend/src/services/risk.ts
import { apiClient } from './api';
import { RiskAssessment } from '../types';

export interface ModelInfo {
  model_type: string;
  features: string[];
  classes: string[];
  validation_accuracy: number;
  feature_importances: Record<string, number>;
  is_demo: boolean;
  notice: string;
}

export const fetchModelInfo = async (): Promise<ModelInfo> => {
  const response = await apiClient.get<ModelInfo>('/risk/model/info');
  return response.data;
};

export const fetchChangeRisk = async (changeId: string): Promise<RiskAssessment> => {
  const response = await apiClient.get<RiskAssessment>(`/risk/${changeId}`);
  return response.data;
};

export const evaluateRisk = async (rawContent?: string, planJson?: any): Promise<RiskAssessment> => {
  const response = await apiClient.post<RiskAssessment>('/risk/analyze', {
    raw_content: rawContent,
    plan_json: planJson,
  });
  return response.data;
};
